policy/test_overlayfs.te

```diff
@@ -0,0 +1,169 @@
+manage_sock_files_pattern(test_overlay_mounter_t, test_overlay_files_rwx_t, test_overlay_files_rwx_t)
+
+manage_files_pattern(test_overlay_mounter_t, test_overlay_whiteout_files_t, test_overlay_whiteout_files_t)
+
+#
+# Mounter needs to search test_file_t directory and execute tests
+#
+allow test_overlay_mounter_t test_file_t:dir search_dir_perms;
+can_exec(test_overlay_mounter_t, test_file_t)
+
+#
+# Client policy
+#
+allow test_overlay_client_t self:dir list_dir_perms;
+allow test_overlay_client_t self:file read_file_perms;
+
+unconfined_runs_test(test_overlay_client_t)
+mcs_constrained(test_overlay_client_t)
+
+corecmd_shell_entry_type(test_overlay_client_t)
+corecmd_entrypoint_all_executables(test_overlay_client_t)
+corecmd_exec_bin(test_overlay_client_t)
+
+kernel_search_proc(test_overlay_client_t)
+kernel_read_proc_symlinks(test_overlay_client_t)
+
+userdom_read_inherited_user_tmp_files(test_overlay_client_t)
+userdom_search_admin_dir(test_overlay_client_t)
+userdom_search_user_home_content(test_overlay_client_t)
+
+fs_getattr_xattr_fs(test_overlay_client_t)
+
+selinux_getattr_fs(test_overlay_client_t)
+
+#
+# client should be allowed to search/read/execute r/o directories and files
+#
+read_files_pattern(test_overlay_client_t, test_overlay_files_ro_t, test_overlay_files_ro_t)
+list_dirs_pattern(test_overlay_client_t, test_overlay_files_ro_t, test_overlay_files_ro_t)
+read_lnk_files_pattern(test_overlay_client_t, test_overlay_files_ro_t, test_overlay_files_ro_t)
+read_chr_files_pattern(test_overlay_client_t, test_overlay_files_ro_t, test_overlay_files_ro_t)
+read_sock_files_pattern(test_overlay_client_t, test_overlay_files_ro_t, test_overlay_files_ro_t)
+can_exec(test_overlay_client_t, test_overlay_files_ro_t)
+
+#
+# Client should be allowed to execute/write/search/read r/w/x directories and files
+#
+manage_files_pattern(test_overlay_client_t, test_overlay_files_rwx_t, test_overlay_files_rwx_t)
+manage_dirs_pattern(test_overlay_client_t, test_overlay_files_rwx_t, test_overlay_files_rwx_t)
+manage_lnk_files_pattern(test_overlay_client_t, test_overlay_files_rwx_t, test_overlay_files_rwx_t)
+rw_chr_files_pattern(test_overlay_client_t, test_overlay_files_rwx_t, test_overlay_files_rwx_t)
+manage_sock_files_pattern(test_overlay_client_t, test_overlay_files_rwx_t, test_overlay_files_rwx_t)
+can_exec(test_overlay_client_t, test_overlay_files_rwx_t)
+allow test_overlay_client_t test_overlay_files_rwx_t:file { relabelfrom relabelto };
+
+#
+# Client needs to manage test_overlay_whiteout_files_t and transition on file named "whiteout"
+#
+manage_files_pattern(test_overlay_client_t, test_overlay_whiteout_files_t, test_overlay_whiteout_files_t)
+manage_dirs_pattern(test_overlay_client_t, test_overlay_whiteout_files_t, test_overlay_whiteout_files_t)
+filetrans_pattern(test_overlay_client_t, test_overlay_files_rwx_t, test_overlay_whiteout_files_t, file, "whiteout")
+
```

Comment:
Misnamed files, should have been transition_files_t, they are files created with a transition rule.
Originally created the type for whiteout files, but still have not built the tests for those.